packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -138,6 +138,14 @@ async function downloadRawFile(
 	return response?.data ?? null;
 }
 
+async function downloadFile(datasetId: string, filename: string): Promise<string | null> {
```

Comment:
I don't think this would work like that... `.../download-file` isn't returning a string, and then that string is treated like a url. That would be the `/download-url` endpoint we were discussing on the call no?

I believe that this endpoint is returning an `ArrayBuffer` like seen in `artifact.ts`?